### PR TITLE
[CARBONDATA-1618]Table comment support for alter table

### PR DESCRIPTION
**Pull request by pawanmalwal**
Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [X] Any interfaces changed?
 None
 - [X] Any backward compatibility impacted?
 None
 - [X] Document update required?
None
 - [X] Testing done
        Test case added - TestAlterTableWithTableComment.scala
       
 - [X] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Can one of the admins verify this patch?

**Comment by ravipesala** [comment]
Can one of the admins verify this patch?

**Comment by ravipesala** [comment]
add to whitelist

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/schema/AlterTableSetCommand.scala [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                                             val properties: Map[String, String],
+                                             val isView: Boolean)
+  extends RunnableCommand with SchemaProcessCommand {
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    processSchema(sparkSession)
+  }
+
+  override def processSchema(sparkSession: SparkSession): Seq[Row] = {
+    val LOGGER: LogService = LogServiceFactory.getLogService(this.getClass.getCanonicalName)
```
Comment (ravipesala) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/schema/AlterTableSetCommand.scala]: Refactor the code to extract common code from both commands.

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/CarbonDescribeFormattedCommand.scala [hunk 2]
```diff
@@ -105,6 +105,10 @@ private[sql] case class CarbonDescribeFormattedCommand(
     results ++= Seq(("Table Name: ", relation.tableMeta.carbonTableIdentifier.getTableName, ""))
     results ++= Seq(("CARBON Store Path: ", relation.tableMeta.storePath, ""))
     val carbonTable = relation.tableMeta.carbonTable
+    // Carbon table support table comment
+    val tableComment = carbonTable.getTableInfo.getFactTable.getTableProperties
+      .getOrDefault("comment", "")
```
Comment (ravipesala) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/CarbonDescribeFormattedCommand.scala]: Move the `comment` as constant to CarbonCommonConstants

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/852/

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/schema/AlterTableSetCommand.scala (hunk 1 above)
Comment (pawanmalwal) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/schema/AlterTableSetCommand.scala]: Handled the comment. Please review.

Comment (pawanmalwal) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/schema/AlterTableSetCommand.scala]: Handled the comment please review.

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/CarbonDescribeFormattedCommand.scala (hunk 2 above)
Comment (pawanmalwal) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/CarbonDescribeFormattedCommand.scala]: Done.

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java [hunk 3]
```diff
@@ -1374,6 +1374,12 @@
   @CarbonProperty
   public static final String BITSET_PIPE_LINE = "carbon.use.bitset.pipe.line";
 
+  /**
+   * this will be used to provide comment for table
+   */
+  @CarbonProperty
```
Comment (ravipesala) [inline core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]: its not a carbon property, it is just a constant

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/867/

File: integration/spark2/src/main/scala/org/apache/spark/util/AlterTableUtil.scala [hunk 4]
```diff
@@ -317,4 +321,84 @@ object AlterTableUtil {
...
+        .fromWrapperToExternalTableInfo(wrapperTableInfo, dbName, tableName)
+      val tblPropertiesMap: mutable.Map[String, String] =
+        thriftTable.fact_table.getTableProperties.asScala
+      if (set) {
+        //       This overrides old properties and update the comment parameter of thriftTable
+        //       with the newly added/modified comment since thriftTable also holds comment as its
+        //       direct property.
+
+        properties.foreach {
+          x =>
```
Comment (ravipesala) [inline integration/spark2/src/main/scala/org/apache/spark/util/AlterTableUtil.scala]: move this to above line like properties.foreach { x =>

File: integration/spark2/src/main/scala/org/apache/spark/util/AlterTableUtil.scala [hunk 5]
```diff
@@ -317,4 +321,84 @@ object AlterTableUtil {
...
+            if (x._1.equalsIgnoreCase(CarbonCommonConstants.TABLE_COMMENT)) {
+              tblPropertiesMap.put(x._1, x._2)
+            }
+        }
+      } else {
+        // This removes the comment parameter from thriftTable
+        // since thriftTable also holds comment as its property.
+        propKeys.foreach {
+          x =>
```
Comment (ravipesala) [inline integration/spark2/src/main/scala/org/apache/spark/util/AlterTableUtil.scala]: move this to above line like properties.foreach { x =>

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1483/

File: integration/spark2/src/main/scala/org/apache/spark/util/AlterTableUtil.scala (hunk 5 above)
Comment (pawanmalwal) [reply integration/spark2/src/main/scala/org/apache/spark/util/AlterTableUtil.scala]: Done.

File: integration/spark2/src/main/scala/org/apache/spark/util/AlterTableUtil.scala (hunk 4 above)
Comment (pawanmalwal) [reply integration/spark2/src/main/scala/org/apache/spark/util/AlterTableUtil.scala]: Done.

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java (hunk 3 above)
Comment (pawanmalwal) [reply core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]: Done.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/871/

**Comment by ravipesala** [comment]
@pawanmalwal Please rebase the code, there is one conflict.

**Comment by pawanmalwal** [comment]
@ravipesala I have resolved the conflict. Due to that SDV had failed I guess.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/875/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1489/

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**